esp-hal/MIGRATING-0.21.md

```diff
@@ -97,12 +99,129 @@ The `with_timeout` constructors have been removed in favour of `set_timeout` or
 +let i2c = I2c::new(peripherals.I2C0, io.pins.gpio4, io.pins.gpio5, 100.kHz()).with_timeout(timeout);
 ```
 
-## Changes to half-duplex SPI
+### I2C drivers can now be configured using `i2c::master::Config`
```

Comment:
This diff looks quite goofy, but I'm trusting that you've just been rearranging some things and that it is correct.

Reply:
Yeah I'm not that confident. I couldn't decide how to organize things, and I think per-peripheral is better than per-topic (e.g. config changes, constructor changes, etc.).